File: observation/observation.go
package observation

import (
	"github.com/kamva/mgm/v3"
)

type AssetContentType string

const (
	ImageJpeg AssetContentType = "image/jpeg"
	ImagePng  AssetContentType = "image/png"
)

type ObservationRequestAsset struct {
	ContentType AssetContentType `json:"contentType" bson:"contentType"`
	Data        []byte           `json:"data" bson:"data"`
}

type ObservationRequest struct {
	Assets     []ObservationRequestAsset `json:"assets" bson:"assets"`
	Attributes []string                  `json:"attributes" bson:"attributes"`
	Comment    string                    `json:"comment" bson:"comment"`
	DeviceId   string                    `json:"deviceId" bson:"deviceId"`
	Position   [2]float64                `json:"position" bson:"position"`
	Timestamp  int64                     `json:"timestamp" bson:"timestamp"`
}

type ObservationStatus string

const (
	Pending  ObservationStatus = "pending"
	Rejected ObservationStatus = "rejected"
	Valid    ObservationStatus = "valid"
)

type ReportedObservation struct {
	mgm.DefaultModel `bson:",inline"`

	Status      ObservationStatus `json:"status" bson:"status"`
	Observation ObservationRequest
}

File: AmavmVdhApi/Models/Observation.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System.Text.Json.Serialization;
using System.ComponentModel;

namespace AmavmVdhApi.Models;
public enum ObservationStatus
{
  Pending,
  Rejected,
  Valid,
}

public enum AssetContentType
{
  [Description("image/jpeg")]
  ImageJpeg,

  [Description("image/png")]
  ImagePng,
}

public class ReportedObservation
{
  [BsonId]
  [BsonRepresentation(BsonType.ObjectId)]
  public string? Id { get; set; }

  [BsonElement("status")]
  [JsonPropertyName("status")]
  public ObservationStatus Status { get; set; }

  public ObservationRequest Observation { get; set; } = null!;
}

public class ObservationRequest
{
  [BsonElement("assets")]
  [JsonPropertyName("assets")]
  public ObservationRequestAsset[]? Assets { get; set; } = null!;

  [BsonElement("attributes")]
  [JsonPropertyName("attributes")]
  public string[]? Attributes { get; set; } = null!;

  [BsonElement("comment")]
  [JsonPropertyName("comment")]
  public string? Comment { get; set; } = null!;

  [BsonElement("deviceId")]
  [JsonPropertyName("deviceId")]
  public string DeviceId { get; set; } = null!;

  [BsonElement("position")]
  [JsonPropertyName("position")]
  public double[] Position { get; set; } = null!;

  [BsonElement("timestamp")]
  [JsonPropertyName("timestamp")]
  public DateTimeOffset Timestamp { get; set; }
}

public class ObservationRequestAsset
{
  [BsonElement("contentType")]
  [JsonPropertyName("contentType")]
  public AssetContentType ContentType { get; set; }

  [BsonElement("data")]
  [JsonPropertyName("data")]
  public byte[] Data { get; set; } = null!;
}
